kzg_prover/examples/chunked_univariate_grand_sum.rs

```diff
@@ -0,0 +1,183 @@
+    },
+    cryptocurrency::Cryptocurrency,
+    entry::Entry,
+    utils::{
+        amortized_kzg::{commit_kzg, create_naive_kzg_proof, verify_kzg_proof},
+        big_uint_to_fp, parse_csv_to_entries,
+    },
+};
+
+const K: u32 = 9;
+const N_CURRENCIES: usize = 2;
+const N_USERS_TOTAL: usize = 64;
+const N_USERS_CHUNK: usize = N_USERS_TOTAL / 2;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let path = "../csv/entry_64.csv";
+
+    let mut entries: Vec<Entry<N_CURRENCIES>> = vec![Entry::init_empty(); N_USERS_TOTAL];
+    let mut cryptos = vec![Cryptocurrency::init_empty(); N_CURRENCIES];
+
+    parse_csv_to_entries::<&str, N_CURRENCIES>(path, &mut entries, &mut cryptos).unwrap();
+
+    // Calculate total for all balance entries
+    let mut csv_total: Vec<BigUint> = vec![BigUint::from(0u32); N_CURRENCIES];
+
+    for entry in &entries {
+        for (i, balance) in entry.balances().iter().enumerate() {
+            csv_total[i] += balance;
+        }
+    }
+
+    // Split the user base into two equal chunks of N_USERS_TOTAL/2 each
+    let entries_first_chunk = entries[0..N_USERS_CHUNK].to_vec();
+    // Calculate the total for the first chunk
+    let mut csv_total_1: Vec<BigUint> = vec![BigUint::from(0u32); N_CURRENCIES];
+    for entry in &entries_first_chunk {
+        for (i, balance) in entry.balances().iter().enumerate() {
+            csv_total_1[i] += balance;
+        }
+    }
+    let entries_second_chunk = entries[N_USERS_CHUNK..].to_vec();
+    // Calculate the total for the second chunk
+    let mut csv_total_2: Vec<BigUint> = vec![BigUint::from(0u32); N_CURRENCIES];
+    for entry in &entries_second_chunk {
+        for (i, balance) in entry.balances().iter().enumerate() {
+            csv_total_2[i] += balance;
+        }
+    }
+
+    assert!(
+        &csv_total_1[BALANCES_INDEX - 1] + &csv_total_2[BALANCES_INDEX - 1]
+            == csv_total[BALANCES_INDEX - 1],
+        "The sum of the chunks' total should be equal to the grand total"
+    );
+
+    type CONFIG = NoRangeCheckConfig<N_CURRENCIES, N_USERS_CHUNK>;
+
+    let circuit_1 = UnivariateGrandSum::<N_USERS_CHUNK, N_CURRENCIES, CONFIG>::init_empty();
+    // Generate the setup artifacts using an empty circuit
+    let (params, pk, vk) = generate_setup_artifacts(K, None, &circuit_1).unwrap();
+
+    // Instantiate the actual circuits for the first and second chunk
+    let circuit_1 =
+        UnivariateGrandSum::<N_USERS_CHUNK, N_CURRENCIES, CONFIG>::init(entries_first_chunk);
+    let circuit_2 =
+        UnivariateGrandSum::<N_USERS_CHUNK, N_CURRENCIES, CONFIG>::init(entries_second_chunk);
+
+    // The zkSNARK proofs encode the balances of the first chunk and the second chunk
+    // in the corresponding advice polynomials
+    let (proof_1, advice_polys_1, _) = full_prover(&params, &pk, circuit_1.clone(), &[vec![]]);
+    let (proof_2, advice_polys_2, _) = full_prover(&params, &pk, circuit_2.clone(), &[vec![]]);
+
+    const BALANCES_INDEX: usize = 1;
```

Comment:
Consider moving the declaration of `BALANCES_INDEX` to line 68, as it is first referenced there.